### PR TITLE
AIPluginTool: reduce token usage, fix context window error

### PR DESCRIPTION
**Pull request by ngundotra**
Using the AIPluginTool fails when the targeted ChatGPT Plugin server's YAML has more than 5 well-documented endpoints.
This causes the context length to overflow for most LLMs very quickly while using AIPluginTool. (See #2140 ).

This PR introduces 2 changes to AIPluginTool to make it more usable.

1. Instead of returning the entire YAML file of the plugin, only return the parsed schemas necessary to execute get & post requests against the plugin. I've chosen to represent the schema in a human readable way, (e.g. "/myEndpoint: (post) my description. Request body: message (string required)". However this should probably be configurable by developers going forward.

2. Instead of using the YAML to construct proper HTTP requests using the `requests_*` tools, I've changed the tool to accept `get` and `post` requests against the endpoints described by the YAML. This reduces the token length needed to communicate with the plugin as well.

Let me know what you think & how I can improve this PR! 🥂

### CODE REVIEW

**Comment by ngundotra** [comment]
Can I please get some assistance with lint / formatting?

**Comment by ngundotra** [comment]
Fixed the formatting issues! @hwchase17 wdyt?

File: langchain/tools/plugin.py [hunk 1]
```diff
@@ -45,37 +49,125 @@ def marshal_spec(txt: str) -> dict:
...
+                if info.get:
+                    operations[(path, "get")] = APIOperation.from_openapi_spec(
+                        openapi_schema, path, "get"
+                    )
+                if info.post:
+                    operations[(path, "post")] = APIOperation.from_openapi_spec(
+                        openapi_schema, path, "post"
+                    )
+
+        print("Description: ", description)
```
Comment (hwchase17) [inline langchain/tools/plugin.py:106]: stray print

**Review by hwchase17 — COMMENTED** [review]
this is great, but i think this should actually be a separate tool!

eg no need to make the usage_guide part of this

i would also add notebook on how to use. can just be example of creating tool and then calling with inputs (doesnt even have to be hooked up to llm)

i have some cycles tonight, can look into

**Comment by cmoski** [comment]
One issue with this particular PR is that if you are including request parameters in post requests (as opposed to in data in the POST request), it doesn't seem to work.

Example of AI Plugin where its failing:
https://gist.github.com/cmoski/d5f9367ebfff5ce79e3e912cfca54829

**Comment by baskaryan** [comment]
stale afaict, closing for now but let me know if i'm missing something!

**Pull request closed without merging**